Sources/SourceKit/SourceKitServer.swift

```diff
@@ -216,6 +216,32 @@ public final class SourceKitServer: LanguageServer {
   }
 }
 
+// MARK: - Build System Delegate
+
+extension SourceKitServer: BuildSystemDelegate {
+  public func fileBuildSettingsChanged(_ changedFiles: Set<URL>) {
+    guard let workspace = self.workspace else {
+      return
+    }
+    let documentManager = workspace.documentManager
+    let openedDocuments = documentManager.openDocuments()
```

Comment:
What's the reasoning for iterating over all documents instead of iterating over the changed files?

Reply:
Changed